fprime-zephyr/Os/Mutex.cpp

```diff
@@ -0,0 +1,48 @@
+// ======================================================================
+// \title Os/Zephyr/Mutex.cpp
+// \brief Zephyr implementation for Os::Mutex
+// Relevant Documentation:
+// https://docs.zephyrproject.org/latest/kernel/services/synchronization/mutexes.html
+// https://docs.zephyrproject.org/latest/doxygen/html/group__mutex__apis.html
+// ======================================================================
+#include <Fw/Types/Assert.hpp>
+#include "fprime-zephyr/Os/Mutex.hpp"
+
+namespace Os {
+namespace Zephyr {
+namespace Mutex {
+
+ZephyrMutex::ZephyrMutex() : Os::MutexInterface(), m_handle() {
+    // set attributes
+    k_mutex_init(&this->m_handle.m_mutex_descriptor);

```

Comment:
Assert return == 0 (zephyr says this is the only return option), however; if that changes we want to know.